lib/std/terminfo.zig

```diff
@@ -0,0 +1,1068 @@
+    .{ .short = "devt", .long = .{ .string = .device_type } },
+    .{ .short = "dispc", .long = .{ .string = .display_pc_char } },
+    .{ .short = "endbi", .long = .{ .string = .end_bit_image_region } },
+    .{ .short = "smpch", .long = .{ .string = .enter_pc_charset_mode } },
+    .{ .short = "smsc", .long = .{ .string = .enter_scancode_mode } },
+    .{ .short = "rmpch", .long = .{ .string = .exit_pc_charset_mode } },
+    .{ .short = "rmsc", .long = .{ .string = .exit_scancode_mode } },
+    .{ .short = "getm", .long = .{ .string = .get_mouse } },
+    .{ .short = "kmous", .long = .{ .string = .key_mouse } },
+    .{ .short = "minfo", .long = .{ .string = .mouse_info } },
+    .{ .short = "pctrm", .long = .{ .string = .pc_term_options } },
+    .{ .short = "pfxl", .long = .{ .string = .pkey_plab } },
+    .{ .short = "reqmp", .long = .{ .string = .req_mouse_pos } },
+    .{ .short = "scesc", .long = .{ .string = .scancode_escape } },
+    .{ .short = "s0ds", .long = .{ .string = .set0_des_seq } },
+    .{ .short = "s1ds", .long = .{ .string = .set1_des_seq } },
+    .{ .short = "s2ds", .long = .{ .string = .set2_des_seq } },
+    .{ .short = "s3ds", .long = .{ .string = .set3_des_seq } },
+    .{ .short = "setab", .long = .{ .string = .set_a_background } },
+    .{ .short = "setaf", .long = .{ .string = .set_a_foreground } },
+    .{ .short = "setcolor", .long = .{ .string = .set_color_band } },
+    .{ .short = "smglr", .long = .{ .string = .set_lr_margin } },
+    .{ .short = "slines", .long = .{ .string = .set_page_length } },
+    .{ .short = "smgtb", .long = .{ .string = .set_tb_margin } },
+    .{ .short = "ehhlm", .long = .{ .string = .enter_horizontal_hl_mode } },
+    .{ .short = "elhlm", .long = .{ .string = .enter_left_hl_mode } },
+    .{ .short = "elohlm", .long = .{ .string = .enter_low_hl_mode } },
+    .{ .short = "erhlm", .long = .{ .string = .enter_right_hl_mode } },
+    .{ .short = "ethlm", .long = .{ .string = .enter_top_hl_mode } },
+    .{ .short = "evhlm", .long = .{ .string = .enter_vertical_hl_mode } },
+    .{ .short = "sgr1", .long = .{ .string = .set_a_attributes } },
+    .{ .short = "slength", .long = .{ .string = .set_pglen_inch } },
+};
+
+pub fn getCapsLookUpTable(allocator: *std.mem.Allocator) !std.StringHashMap(Terms) {
+    var data = std.StringHashMap(Terms).init(allocator);
+    for (dictionary) |term| 
+        _ = try data.put(term.short, term.long);
+    return data;
+}
+
+pub const Terminfo = struct {
+    names: std.StringHashMap(bool),
+    caps: std.StringHashMap([]const u8),
+
+    pub fn init(allocator: *std.mem.Allocator, terminfo_path: []const u8, cap_delimiter: []const u8) !Terminfo {
+        var names = std.StringHashMap(bool).init(allocator);
+        var caps = std.StringHashMap([]const u8).init(allocator);
+
+        var read_line = try std.fs.ReadLineIterator.init(terminfo_path);
+        defer read_line.deinit();
+        read_line.setFilter(struct {
```

Comment:
this could be simplified to:
```zig
.{
    fn filter(line: []u8) bool {
        return !(std.mem.startsWith(u8, line, "#") or line.len == 0);
    }
}.filter
```